Automattic/jetpack
# [Super Cache] Prevent deleting freshly preloaded files when GC time set to 0

Fixes https://wordpress.org/support/topic/preload-mode-skips-caching-pages-or-deletes-already-cached-pages

When preloads finish, `prune_super_cache` is called to clean out leftovers. Unfortunately, if `$cache_max_time` is set to 0, preload will delete any files at least one second old - which frequently includes preloaded files. It causes intermittent post-preload disappearances.

This PR fixes this by ensuring the prune process will not delete files based on mtime if `$cache_max_time` is 0.

## Proposed changes:
* Prevent prune_super_cache from deleting files if `$cache_max_time` is set to 0.

### Other information:

## Jetpack product discussion


n/a

## Does this pull request change what data or activity we track or use?


no

## Testing instructions:
* Set cache timeout to 0
* Manually preload over and over
* Without this patch, sometimes files created near the start of the preload will get deleted at the end
* With this patch, all files created during the preload should remain. :)
